aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+                # if the scene is implicit just assign it name Scene:0
+                if len(scene_names) < 1:
+                    scene_names = ["Scene:0"]
+                self._scenes = tuple(scene_names)
+
+        return self._scenes
+
+    @staticmethod
+    def _get_image_data(
+        fs: AbstractFileSystem,
+        path: str,
+        scene: int,
+        retrieve_dims: List[str],
+        retrieve_indices: List[Optional[int]],
+    ) -> np.ndarray:
+        """
+        Open a file for reading, construct a Zarr store, select data, and compute to
+        numpy.
+
+        Parameters
+        ----------
+        fs: AbstractFileSystem
+            The file system to use for reading.
+        path: str
+            The path to file to read.
+        scene: int
+            The scene index to pull the chunk from.
+        retrieve_dims: List[str]
+            The order of the retrieve indicies operations
+        retrieve_indices: List[Optional[int]],
+            The image index operations to retrieve.
+            If None, retrieve the whole dimension.
+
+        Returns
+        -------
+        chunk: np.ndarray
+            The image chunk as a numpy array.
+        """
+        # Open and select the target image
+        with fs.open(path) as open_resource:
+            czi = CziFile(open_resource)
+            dims_shape = CziReader._dims_shape_to_scene_dims_shape(
+                dims_shape=czi.get_dims_shape(),
+                scene_index=scene,
+                consistent=czi.shape_is_consistent,
+            )
+
+            # Create the fill array shape
+            # Drop the YX as we will be pulling the individual YX planes
+            retrieve_shape: List[int] = []
+            use_selected_or_np_map: Dict[str, Optional[int]] = {}
+            for dim, index_op in zip(retrieve_dims, retrieve_indices):
+                if (
+                    dim not in [DimensionNames.SpatialY, DimensionNames.SpatialX, "A"]
+                    and dim in dims_shape.keys()
+                ):
+                    # Handle slices
+                    if index_op is None:
+                        # Store the dim for later to inform to use the np index
+                        use_selected_or_np_map[dim] = None
+                        if dim == DimensionNames.MosaicTile:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.MosaicTile][1]
+                            )
+                        elif dim == DimensionNames.Time:
+                            retrieve_shape.append(dims_shape[DimensionNames.Time][1])
+                        elif dim == DimensionNames.Channel:
+                            retrieve_shape.append(dims_shape[DimensionNames.Channel][1])
+                        elif dim == DimensionNames.SpatialZ:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.SpatialZ][1]
```

Comment:
Just to be sure - the `[1]`s here are intentional? They aren't supposed to be `[0]`?

Reply:
Yes the key points to a (start, length) pair and we only care about the length.